File: frontend/src/api/embyLibraries.ts
import request from '@/utils/request'

export function listEmbyLibraries(serverId?: string) {
  return request({
    url: '/api/emby-libraries/list',
    method: 'get',
    params: { server_id: serverId }
  })
}

export function addEmbyLibrary(name: string, collectionType: string, path?: string, serverId?: string) {
  return request({
    url: '/api/emby-libraries/add',
    method: 'post',
    params: { name, collection_type: collectionType, path, server_id: serverId }
  })
}

export function updateEmbyLibrary(libraryData: any, server_id?: string) {
  return request({
    url: '/api/emby-libraries/update',
    method: 'post',
    data: libraryData,
    params: { server_id }
  })
}

export function removeEmbyLibrary(name: string, id: string, serverId?: string) {
  return request({
    url: '/api/emby-libraries/remove',
    method: 'delete',
    params: { name, id, server_id: serverId }
  })
}

File: backend/app/api/emby_libraries.py
from fastapi import APIRouter, HTTPException, Query
from typing import List, Dict, Any, Optional
from app.services.emby_library_service import get_emby_library_service

router = APIRouter()

@router.get("/list")
async def list_libraries(server_id: Optional[str] = None):
    service = get_emby_library_service(server_id)
    if not service:
        raise HTTPException(status_code=400, detail="Emby server not configured")
    return await service.get_libraries()

@router.post("/add")
async def add_library(name: str, collection_type: str, path: Optional[str] = None, server_id: Optional[str] = None):
    service = get_emby_library_service(server_id)
    if not service:
        raise HTTPException(status_code=400, detail="Emby server not configured")
    success = await service.add_library(name, collection_type, path)
    if not success:
        raise HTTPException(status_code=500, detail="Failed to add library")
    return {"message": "Library added"}

@router.post("/update")
async def update_library(library_data: Dict[str, Any], server_id: Optional[str] = None):
    service = get_emby_library_service(server_id)
    if not service:
        raise HTTPException(status_code=400, detail="Emby server not configured")
    success = await service.update_library_options(library_data)
    if not success:
        raise HTTPException(status_code=500, detail="Failed to update library")
    return {"message": "Library updated"}

@router.delete("/remove")
async def remove_library(name: str, id: str, server_id: Optional[str] = None):
    service = get_emby_library_service(server_id)
    if not service:
        raise HTTPException(status_code=400, detail="Emby server not configured")
    success = await service.delete_library(name, id)
    if not success:
        raise HTTPException(status_code=500, detail="Failed to remove library")
    return {"message": "Library removed"}